ast/compile_test.go

```diff
@@ -701,6 +699,52 @@ p2 := 2`})
 	assertCompilerErrorStrings(t, c, expected)
 }
 
+func TestCompilerCheckUndefinedFuncs(t *testing.T) {
+
+	module := `
+		package test
+
+		undefined_function {
+			data.deadbeef(x)
+		}
+
+		undefined_global {
+			deadbeef(x)
+		}
+
+		undefined_dynamic_dispatch {
+			x = "f"; data.test2[x](1)  # not currently supported
+		}
+	`
+
+	module2 := `
```

Comment:
nit: Is `module2` used for anything? Asking more out of curiosity to make sure I'm understanding the code 😄